File: DSA/VivaswanSingh_590023022/Day_18/q2.c
struct ListNode* removeNthFromEnd(struct ListNode* head, int n)
{
    struct ListNode dummy;
    dummy.next = head;

    struct ListNode *fast = &dummy;
    struct ListNode *slow = &dummy;

    for (int i = 0; i <= n; i++)
    {
        fast = fast->next;
    }

    while (fast != NULL)
    {
        fast = fast->next;
        slow = slow->next;
    }

    struct ListNode *nodeToDelete = slow->next;
    slow->next = nodeToDelete->next;
    free(nodeToDelete);

    return dummy.next;
}

File: DSA/Navneet_590012555/Day18/Q2.java
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode d = new ListNode(0);
        d.next = head;

        ListNode f = d, s = d;

        for (int i = 0; i <= n; i++)
            f = f.next;

        while (f != null) {
            f = f.next;
            s = s.next;
        }

        s.next = s.next.next;

        return d.next;
    }
}